src/ui/public/visualize/visualize.js

```diff
@@ -50,9 +52,19 @@ uiModules
       const responseHandler = getHandler(responseHandlers, $scope.vis.type.responseHandler);
 
       $scope.fetch = _.debounce(function () {
+
         // searchSource is only there for courier request handler
         requestHandler($scope.vis, $scope.appState, $scope.uiState, $scope.savedObj.searchSource)
-          .then(resp => responseHandler($scope.vis, resp), e => {
+          .then(requestHandlerResponse => {
+            if ($scope.requestHandlerResponse && $scope.requestHandlerResponse === requestHandlerResponse) {
+              //we do not need to call the response handler, since there have been no data changes
```

Comment:
thats not always the case. response handlers receive vis configuration as well, and they may produce different response based on that.

i don't see a reason for skipping this part, could you elaborate a bit ? all we do is save some resources, which shouldn't be that much anyway ...

if we are to skip it, we should make sure vis object didn't change as well ...